### PR TITLE
Specific error for unsupported data platform feature

### PR DESCRIPTION
**Pull request by DevonFulcher**
### Description

This PR replaces a generic `RuntimeError` for a more specific exception, `UnsupportedEngineFeatureError`. This makes it easier for clients to handle this exception. I ran the unit tests locally, but other than that I didn't do much testing.

### CODE REVIEW

File: metricflow/errors/errors.py
Comment (tlento) [inline metricflow/errors/errors.py:85]: I prefer to avoid engine-specific errors. Shall we generalize this to something like `UnsupportedEngineFeatureError` instead? The message would have to go back to being passed in at the callsite, as it was before.

This also allows us to use the same custom exception for all of the various percentile cases we have to deal with. For example, [Databricks](https://github.com/dbt-labs/metricflow/blob/main/metricflow/sql/render/databricks.py#L27-L35), [Snowflake](https://github.com/dbt-labs/metricflow/blob/main/metricflow/sql/render/snowflake.py#L39-L43), [Redshift](https://github.com/dbt-labs/metricflow/blob/main/metricflow/sql/render/redshift.py#L31-L40), and [Postgres](https://github.com/dbt-labs/metricflow/blob/main/metricflow/sql/render/postgres.py#L63-L72) all have different levels of support for different percentile aggregation configurations, so we should probably throw the same error for each one.

**Review by tlento — COMMENTED** [review]
I love the custom exception, RuntimeError for stuff this specific always bothers me, but I think we a more general error will serve us better.

It's probably worth just an empty custom exception type that replaces RuntimeError in all of the percentile mayhem linked in the inline comment. Then we can update that however we like later on.

Note I'm fine with any error type that isn't so engine-specific here. If you'd prefer to limit this to Percentile issues that's reasonable.

File: metricflow/errors/errors.py
Comment (plypaul) [reply metricflow/errors/errors.py:85]: +1 for `UnsupportedEngineFeatureError`

**Review by plypaul — CHANGES_REQUESTED** [review]
To move this out of my review queue, marking as `Request changes`.

**Review by tlento — APPROVED** [review]
Fantastic! Do you have merge permissions? If so, please use squash and merge with a custom commit message on the squash commit to keep the git log streamlined. Thank you!

**Comment by DevonFulcher** [comment]
Thanks, Tom! I'm not sure if I have merge permissions. Right now, the PR is still blocked by Paul's request for changes. No rush, though!

**Pull request merged**
